usr.bin/ssh/sshd.c

```diff
@@ -718,34 +716,57 @@ pack_config(struct sshbuf *conf)
 	POKE_U32(sshbuf_mutable_ptr(m), len - 4);
 
 	sshbuf_free(inc);
-	sshbuf_free(hostkeys);
 
 	debug3_f("done");
 	return m;
 }
 
+/*
+ * Protocol from reexec master to child:
+ *	uint32  size
+ *	uint8   type (ignored)
+ *	string	configuration
+ *	uint64	timing_secret
+ *	string	included_files[] {
+ *		string	selector
+ *		string	filename
+ *		string	contents
+ *	}
+ * Second message
+ *	uint32  size
+ *	uint8   type (ignored)
+ *	string	host_keys[] {
+ *		string private_key
+ *		string public_key
+ *		string certificate
+ *	}
+ */
 static void
-send_rexec_state(int fd, struct sshbuf *config)
+send_rexec_state(int fd)
```

Comment:
Maybe add a comment here that this function is only called for `debug_mode || inetd_mode` and that the usual case is handled asynchronously by `server_accept_loop()`

Reply:
I've added a comment, but also changed to function to use fork() so the config is sent child proc to exec-ed parent

this avoids the problems w/ the socketpair buffer size

we've been talking about this before, but I think we could add the fork() here (it's only debug/inetd mode)

Reply:
IMO that's fine